packages/babel-plugin-ember-test-metadata/get-normalized-file-path.js

```diff
@@ -1,36 +1,32 @@
-const {
-  _getRelativePathForClassic,
-  _getRelativePathForClassicInRepo,
-  _getRelativePathForEmbroider,
-  _getRelativePathForEmbroiderInRepo,
-} = require('./get-relative-paths');
-
-const path = require('path');
-
 /**
  * Get a normalized file path, based on whether the app build is classic or with Embroider
  * @param {object} opts Babel state.file.opts which include root and filename props
  * @returns {string} E.g. tests/acceptance/my-test.js
  */
-function getNormalizedFilePath({ packageName, isUsingEmbroider, filename, root }) {
-  if (!isUsingEmbroider) {
-    if (filename.includes('ember-add-in-repo-tests')) {
-      return _getRelativePathForClassicInRepo(filename);
-    }
+function getNormalizedFilePath({ packageName, filename, isUsingEmbroider, root, sourceFileName }) {
+  let file = sourceFileName || filename;
 
-    return _getRelativePathForClassic(filename, packageName);
-  } else {
-    const rootDirWithBase = path.join(path.parse(root).dir, path.parse(root).base);
-    if (filename.includes(rootDirWithBase)) {
-      filename = filename.replace(rootDirWithBase, '');
-    }
+  if (file.includes(root)) {
+    const regex = new RegExp(`.*${root}/`);
+    file = file.replace(regex, '');
+  }
 
-    if (filename.includes('ember-add-in-repo-tests')) {
-      return _getRelativePathForEmbroiderInRepo(filename);
-    }
+  if (file.startsWith(`${packageName}/`)) {
+    const regex = new RegExp(`^${packageName}/`);
+    file = file.replace(regex, '');
+  }
 
-    return _getRelativePathForEmbroider(filename);
+  if (isUsingEmbroider) {
+    const regex = new RegExp(`^.*embroider/.{6}/`);
```

Comment:
Given that this is an OSS project, the other benefit of the previous implementation was that it accommodated Windows based file paths. If we do go w/regular expressions, we should use `path.sep` instead of hardcoding the slashes.

Reply:
I need to understand the performance implications of creating so many `RegExp` instances. Since this plugin runs in our build pipeline, we have to be very sensitive to memory or CPU pressure as a result of changes that are added to babel plugins.

Have you done any performance analysis in a large repository related to this change?

Reply:
not for this change; in fact, I don't know that this project has ever had a perf analysis. do you remember differently? How would you want to see this benchmarked?

Reply:
We did some analysis when we originally wrote it to ensure build times weren't going to be adversely affected. I'll have to dig it up, but I believe we used broccoli's debug flags which expose Heimdall counters to determine the overall plugin perf.

eg.

```shell
DEBUG_DEPTH=5 DEBUG_LEVEL=debug DEBUG=broccoli-merge-trees ember build
```